analyzers/src/SonarAnalyzer.Common/Rules/ParameterNameMatchesOriginalBase.cs

```diff
@@ -80,20 +80,42 @@ private void VerifyParameters(SyntaxNodeAnalysisContext context, TMethodDeclarat
             }
         }
 
-        private void VerifyGenericParameters(SyntaxNodeAnalysisContext context, TMethodDeclarationSyntax methodSyntax, IList<IParameterSymbol> actualParameters,
-            IList<IParameterSymbol> expectedParameters, string expectedLocation)
+        private void VerifyGenericParameters(SyntaxNodeAnalysisContext context,
+                                             TMethodDeclarationSyntax methodSyntax,
+                                             IList<IParameterSymbol> actualParameters,
+                                             IList<IParameterSymbol> expectedParameters,
+                                             string expectedLocation)
         {
             var parameters = ParameterIdentifiers(methodSyntax).ToList();
             for (var i = 0; i < parameters.Count; i++)
             {
                 var parameter = parameters[i];
                 var expectedParameter = expectedParameters[i];
                 if (!parameter.ValueText.Equals(expectedParameter.Name, Language.NameComparison)
-                    && (expectedParameter.Type.Kind != SymbolKind.TypeParameter || actualParameters[i].Type.Kind == SymbolKind.TypeParameter))
+                    && !AreGenericWithDifferentTypes(actualParameters[i].Type, expectedParameter.Type)
+                    && (expectedParameter.Type.Kind != SymbolKind.TypeParameter
+                        || actualParameters[i].Type.Kind == SymbolKind.TypeParameter))
                 {
                     context.ReportIssue(Diagnostic.Create(rule, parameter.GetLocation(), parameter.ValueText, expectedParameter.Name, expectedLocation));
                 }
             }
         }
+
+        private static bool AreGenericWithDifferentTypes(ITypeSymbol actualType, ITypeSymbol expectedType)
```

Comment:
I would change this name to something like `AreGenericTypeParametersWithDifferentTypes`. To be a bit more clear what the method does.
WDYT?